### ISSUE TITLE
Extend Elyra tutorial with Neural Magic version.

### ISSUE THREAD

**Issue by pacospace** [AICoE/elyra-aidevsecops-tutorial#297]
**Is your feature request related to a problem? Please describe.**
Extend Elyra tutorial with Neural Magic [1] version.

**High-level Goals**
Elyra tutorial has one version that focuses on the use of Neural Magic.

**Describe the solution you'd like**
https://docs.google.com/document/d/1GNICmV_EcI-PPygxvzTupBCIt9HG9MKiEUeVy6YVO2c/edit#heading=h.fbxyxnppymly

**Describe alternatives you've considered**
n/a

**Additional context**
This is a key result we would like to achieve in 21Q3

Related-To: https://github.com/AICoE/aicoe-ci/issues/120
Related-To: https://github.com/AICoE/aicoe-ci/issues/118

*Tasks*
- [x] Extend docs for Neural Magic
- [x] Create new notebooks
- [x] Add new dataset
- [x] Add new models
- [x] Add Elyra pipeline for Neural Magic.

**Acceptance Criteria**
- [ ] demo extension can be run in Operate First
- [x] STRETCH GOAL: feedback pipeline is able to show results from two deployments at least (Tf+Flask+gunicorn, Pytorch+Flask+gunicorn, neuralmagic+Flask+gunicorn): https://devconfus2021.sched.com/event/lkgM/feedback-about-deployment-of-an-intelligent-app

**References**
- [1] https://neuralmagic.com/

cc @goern @riekrh @TreeinRandomForest

**Comment by goern** [comment]
sounds good to me, thanks for writing this up. please take it to the next Tech Talk session to circulate it with the team

**Comment by pacospace** [comment]
Depends-On: https://github.com/neuralmagic/sparseml/issues/386

**Comment by pacospace** [comment]
Depends-On: https://github.com/neuralmagic/sparseml/issues/387

**Comment by pacospace** [comment]
Related-To: https://github.com/neuralmagic/sparsify/issues/83

**Comment by pacospace** [comment]
I was able to create an example for MNIST Classification using Pytorch + Neural Magic Tools. The output is an MNIST pruned ONNX model optimized with Neural Magic Sparsify Recipe.

The next step will be to deploy that ONNX model with NeuralMagic deepsparse.

**Comment by pacospace** [comment]
Nice-to-Have: https://github.com/neuralmagic/sparsify/issues/84

**Comment by pacospace** [comment]
I was able to deploy deepsparse with MNIST example. I will move to the gather metrics pipeline extension now.

**Comment by pacospace** [comment]
Deployment in the cluster with deepsparse is currently failing because there are no node with flags avx3 or avx512. So I'm blocked on that.

**Comment by pacospace** [comment]
demo is also blocked until we get KFP in smaug.

**Comment by pacospace** [comment]
Depends-On: https://github.com/operate-first/support/issues/409

**Comment by pacospace** [comment]
Neural magic deployment cannot be run on smaug cluster at the moment, I added a note

NOTE: _Deepsparse deployment at the moments works only with CPU that supports avx2, avx512 flags, even bette if [avx512_vnni](https://en.wikichip.org/wiki/x86/avx512_vnni) with VNNI is available. Please check your environment running `cat /proc/cpuinfo` to check flags and verify if your machine supports the deployment. In the deployment manifests in `nm-inference` overlay you need to set the env variable NM_ARCH=avx2|avx512._

**Comment by pacospace** [comment]
> Deployment in the cluster with deepsparse is currently failing because there are no node with flags avx2 or avx512. So I'm blocked on that.

Neural magic deployment cannot be run on smaug cluster at the moment, I added a note

NOTE: _Deepsparse deployment at the moments work only with CPU that supports avx2, avx512 flags (even better if [avx512_vnni](https://en.wikichip.org/wiki/x86/avx512_vnni) with VNNI is available: https://github.com/neuralmagic/deepsparse/issues/186). Please check your environment running `cat /proc/cpuinfo` to identify flags and verify if your machine supports the deployment. In the deployment manifests in `nm-inference` overlay you need to set the env variable NM_ARCH=avx2|avx512._

**Comment by goern** [comment]
what is the status of this?

**Comment by pacospace** [comment]
> what is the status of this?

See comment above :) we are still waiting for KFP in Operate FIrst
